L1Trigger/L1THGCal/src/be_algorithms/HGCalTriggerCellCalibration.cc

```diff
@@ -0,0 +1,42 @@
+#include "L1Trigger/L1THGCal/interface/be_algorithms/HGCalTriggerCellCalibration.h"
+
+//class constructor
+HGCalTriggerCellCalibration::HGCalTriggerCellCalibration(const edm::ParameterSet& conf){
+    
+    const edm::ParameterSet beCodecConfig = conf.getParameterSet("calib_parameters");
+
+    LSB_ = beCodecConfig.getParameter<double>("cellLSB"); 
+    fCperMIP_ee_ = beCodecConfig.getParameter<std::vector<double>>("fCperMIPee");
+    fCperMIP_fh_ = beCodecConfig.getParameter<std::vector<double>>("fCperMIPfh");
+    dEdX_weights_ = beCodecConfig.getParameter<std::vector<double>>("dEdXweights");
+    thickCorr_ = beCodecConfig.getParameter<std::vector<double>>("thickCorr");
+}
+
+void HGCalTriggerCellCalibration::calibrate(l1t::HGCalTriggerCell& trgCell, int cellThickness){
+        HGCalDetId trgdetid(trgCell.detId());
+        int trgCellLayer = trgdetid.layer();
+        int subdet =  (ForwardSubdetector)trgdetid.subdetId();
```

Comment:
Same as above. You don't need to do the int -> ForwarSubdetector -> int conversion.